docs/source/worker.rst

```diff
@@ -218,22 +225,182 @@ usage then the worker will start dumping unused data to disk, even if internal
+At 80% load, the worker's thread pool will stop accepting new tasks.  This
 gives time for the write-to-disk functionality to take effect even in the face
 of rapidly accumulating data.
 
 
 Kill Worker
 ~~~~~~~~~~~
 
-At 95% memory load a worker's nanny process will terminate it.  This is to
+At 95% memory load, a worker's nanny process will terminate it. This is to
 avoid having our worker job being terminated by an external job scheduler (like
-YARN, Mesos, SGE, etc..).  After termination the nanny will restart the worker
+YARN, Mesos, SGE, etc..).  After termination, the nanny will restart the worker
 in a fresh state.
 
 
+Using the dashboard to monitor memory usage
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The dashboard (typically available on port 8787) shows a summary of the overall memory
+usage on the cluster, as well as the individual usage on each worker. It provides
+different memory readings:
+
+process
+    Overall memory used by the worker process (RSS), as measured by the OS
+
+managed
+    This is the sum of the ``sizeof`` of all dask data stored on the worker, excluding
+    spilled data.
+
+unmanaged
+    This is the memory usage that dask is not directly aware of. It is estimated by
+    subtracting managed memory from the total process memory and typically includes:
+
+    - The Python interpreter code, loaded modules, and global variables
+    - Memory temporarily used by running tasks
+    - Dereferenced Python objects that have not been garbage-collected yet
+    - Unused memory that the Python memory allocator did not return to libc through
+      `free`_ yet
+    - Unused memory that the user-space libc `free`_ function did not release to the OS
+      yet (see memory allocators below)
+    - Memory fragmentation
+    - Memory leaks
+
+unmanaged recent
+    Unmanaged memory that has appeared within the last 30 seconds. This is not included
+    in the 'unmanaged' memory measure above. Ideally, this memory should be for the most
+    part a temporary spike caused by tasks' heap use plus soon-to-be garbage collected
+    objects.
+
+    The time it takes for unmanaged memory to transition away from its "recent" state
+    can be tweaked through the ``distributed.worker.memory.recent-to-old-time`` key in
+    the ``~/.config/dask/distributed.yaml`` file. If your tasks typically run for longer
+    than 30 seconds, it's recommended that you increase this setting accordingly.
+
+    By default, :meth:`distributed.Client.rebalance` and
+    :meth:`distributed.scheduler.Scheduler.rebalance` ignore unmanaged recent memory.
+    This behaviour can also be tweaked using the dask config - see the methods'
+    documentation.
+
+spilled
+    managed memory that has been spilled to disk. This is not included in the 'managed'
+    measure above.
+
+The sum of managed + unmanaged + unmanaged recent is equal by definition to the process
+memory.
+
+
+.. _memtrim:
+
+Memory not released back to the OS
```

Comment:
@xhochy I pinged you on another PR recently about explicit malloc trim calls. We discussed this and decided to write this doc page for now. If you have anything to add here, we'd appreciate a short review

Reply:
Thanks for the pointer! Overall this looks good 👍 One could add `mimalloc` as an alternative here, too. We have seen slight performance differences between both in Arrow but I don't think they would have an impact here.